gnd/src/main/java/com/google/android/gnd/repository/OfflineAreaRepository.java

```diff
@@ -140,11 +144,11 @@ private Completable enqueueTileDownloads(OfflineArea area) {
 
   public Completable addAreaAndEnqueue(LatLngBounds bounds) {
     OfflineArea offlineArea =
-        // TODO: Geocode
         OfflineArea.newBuilder()
             .setBounds(bounds)
             .setId(uuidGenerator.generateUuid())
             .setState(State.PENDING)
+            .setName(geocodingManager.getOfflineAreaName(bounds))
```

Comment:
Since `geocodingManager.getOfflineAreaName(bounds)` is an async operation, should it be modeled as a `Single<>`? If not, the UI might block while waiting for `addAreaAndEnqueue()` to finish. It might look something like:

```
return geocodingManager.getOfflineAreaName(bounds)
    .map(areaName => OfflineArea.newBuilder()....setName(areaName).build())   
    .flatMapCompletable(this::enqueueTileDownloads);

Reply:
Good point! The geocoder docs recommend calling it on a separate thread as well. I'll make the change.

Reply:
done! 582b803